zap/src/parser/reports.rs

```diff
@@ -251,6 +257,7 @@ impl Report<'_> {
 			Self::AnalyzeOrNestedOptional { .. } => "3021",
 			Self::AnalyzeRecursiveOr { .. } => "3022",
 			Self::AnalyzeConflictingExport { .. } => "3023",
+			Self::AnalyzeNoStringDataKind { .. } => "3024",
```

Comment:
I'm wondering if we should create a new kind of error for deprications (`4001`). Depends on if we choose to PR the instance syntax changes before the next release.